src/chemistry.cpp

```diff
@@ -0,0 +1,169 @@
+  // TODO(kevin): reaction classes read input options directly in their initialization.
+  numReactions_ = config.numReactions;
+  // std::cout << "number of reactions: " << numReactions_ << std::endl;
+  reactionEnergies_.SetSize(numReactions_);
+  reactionEnergies_ = config.reactionEnergies;
+
+  detailedBalance_.SetSize(numReactions_);
+  detailedBalance_ = config.detailedBalance;
+
+  reactions_.resize(numReactions_);
+  reactantStoich_.SetSize(numSpecies_, numReactions_);
+  productStoich_.SetSize(numSpecies_, numReactions_);
+  equilibriumConstantParams_.SetSize(numReactions_, 3);
+  equilibriumConstantParams_ = 0.0;
+
+  for (int r = 0; r < numReactions_; r++) {
+    // std::cout << "reactions-" << r  << std::endl;
+    for (int mixSp = 0; mixSp < numSpecies_; mixSp++) {
+      int inputSp = (*mixtureToInputMap_)[mixSp];
+      // std::cout << mixSp << " : " << inputSp << std::endl;
+      reactantStoich_(mixSp, r) = config.reactantStoich(inputSp, r);
+      productStoich_(mixSp, r) = config.productStoich(inputSp, r);
+      // std::cout << reactantStoich_(mixSp, r) << " - " << productStoich_(mixSp, r) << std::endl;
+    }
+    switch (config.reactionModels[r]) {
+      case ARRHENIUS: {
+        // std::cout << "reactions-" << r << ": Arrhenius" << std::endl;
+        double A = (config.reactionModelParams[r])[0];
+        double b = (config.reactionModelParams[r])[1];
+        double E = (config.reactionModelParams[r])[2];
+        reactions_[r] = new Arrhenius(A, b, E);
+        // std::cout << A << ", " << b << ", " << E << std::endl;
+      } break;
+      case HOFFERTLIEN: {
+        // std::cout << "reactions-" << r << ": HoffertLien" << std::endl;
+        double A = (config.reactionModelParams[r])[0];
+        double b = (config.reactionModelParams[r])[1];
+        double E = (config.reactionModelParams[r])[2];
+        reactions_[r] = new HoffertLien(A, b, E);
+        // std::cout << A << ", " << b << ", " << E << std::endl;
+      } break;
+    }
+    if (detailedBalance_[r]) {
+      // std::cout << "detail balanced: ";
+      for (int d = 0; d < 3; d++) {
+        equilibriumConstantParams_(r, d) = (config.equilibriumConstantParams[r])[d];
+        // std::cout << equilibriumConstantParams_(r, d);
+      }
+      std::cout << std::endl;
+    }
+  }
+}
+
+Chemistry::~Chemistry() {
+  //   if (mixture_ != NULL) delete mixture_;
+  //   if (mixtureToInputMap_ != NULL) delete mixtureToInputMap_;
+  //   if (speciesMapping_ != NULL) delete speciesMapping_;
+  for (int r = 0; r < numReactions_; r++) {
+    if (reactions_[r] != NULL) delete reactions_[r];
+  }
+}
+
+void Chemistry::computeForwardRateCoeffs(const double T_h, const double T_e, Vector& kfwd) {
+  kfwd.SetSize(numReactions_);
+  kfwd = 0.0;
+
+  for (int r = 0; r < numReactions_; r++) {
+    bool isElectronInvolved = isElectronInvolvedAt(r);
+    kfwd = reactions_[r]->computeRateCoefficient(T_h, T_e, isElectronInvolved);
```

Comment:
It looks like `kfwd` here should be indexed by `r`, right?

Reply:
Resolved at branch `axis_mixture`.

Reply:
This is just an indexing bug right?  If so, it should be fixed as part of this PR (or probably on #119 actually... I've been looking things over here since this PR includes everything in #119) rather than separately.

Reply:
Brought the commit from `axis_mixture`.